Model/lib/wdk/model/questions/queries/geneQueries.xml

```diff
@@ -1788,22 +1788,21 @@ AND 1=0
               SELECT distinct gs.transcript_source_id AS source_id, gs.gene_source_id, ta.project_id,
                     'Y' as matched_result,
                     apidb.tab_to_string(set(cast(COLLECT(distinct gs.evidence_code) AS apidb.varchartab)), ', ') as evidence_code,
---                  listagg(gs.is_go_slim, ',')WITHIN GROUP (ORDER BY gs.is_go_slim) as is_go_slim
                     regexp_replace(apidb.tab_to_string(set(cast(COLLECT(distinct to_char(gs.is_go_slim)) AS apidb.varchartab)), ', '),
                                      ', $', '')
                       as is_go_slim
               FROM ApidbTuning.TranscriptAttributes ta, ApidbTuning.GoTermSummary gs
               WHERE ta.taxon_id IN ($$organism$$)
                 AND ta.aa_sequence_id = gs.aa_sequence_id
                 AND decode(gs.evidence_code, 'IEA', 'Computed', 'Curated') in ($$go_term_evidence$$)
---              and case when $$go_term_slim$$ = 'Yes' and gs.is_go_slim = '1' then 1 
---                       when ($$go_term_slim$$ = 'No' and (gs.is_go_slim = '1' or gs.is_go_slim = '0')) then 1 
---                       else 0
- --                 end = 1 
-                AND (gs.go_id IN ($$go_typeahead$$)
-                     OR (lower(gs.go_term_name) like lower(REPLACE($$go_term$$, '*', '%'))
-                     OR lower( gs.go_id) like lower(REPLACE($$go_term$$, '*', '%'))
-                     OR lower(gs.go_id || ' : ' || gs.go_term_name) like lower(REPLACE($$go_term$$, '*', '%'))) )
+                AND (
+                  gs.go_id IN ($$go_typeahead$$) OR (
+                    $$go_term$$ != 'N/A' AND (               
+                      lower(gs.go_term_name) like lower(REPLACE($$go_term$$, '*', '%')) OR
+                      gs.go_id = upper($$go_term$$)
+                    )
+                  )
+                )
```

Comment:
This jibes with what we discussed when we looked at this query together. I think it makes sense not to include GO IDs when checking what a wildcarded input expands to match, and I would expect (and imagine you have confirmed) that explicitly checking whether the $$go_term$$ param is set to "N/A" (meaning the user didn't supply a value for it) greatly improves the performance in those cases.